### PR TITLE
we now pass whole objects into components

### PR DESCRIPTION
**Pull request by shortridge11**

### CODE REVIEW

File: src/assets/library.json [hunk 1]
```diff
@@ -3,533 +3,669 @@
...
       "tier": 3,
       "attack": 6,
-      "health": 3
+      "health": 3,
+      "xp": 0
     },
     {
       "name": "dolphin",
       "display": "Dolphin",
+      "image": "ODolphintter",
```
Comment (prindle-jason) [inline src/assets/library.json]: Thwrongat's.

File: src/assets/library.json [hunk 2]
```diff
@@ -3,533 +3,669 @@
     {
       "name": "duck",
       "display": "Duck",
+      "image": "Duck",
```
Comment (prindle-jason) [inline src/assets/library.json]: All image files were updated to camelCase.  "duck.webp"

File: src/assets/library.json [hunk 3]
```diff
@@ -3,533 +3,669 @@
...
       "tier": 6,
       "attack": 4,
-      "health": 5
+      "health": 5,
+      "xp": 0
     },
     {
       "name": "snake",
       "display": "snake",
+      "image": "snake",
```
Comment (prindle-jason) [inline src/assets/library.json]: Nailed it!

File: src/assets/library.json [hunk 4]
```diff
@@ -3,533 +3,669 @@
...
       "display": "Honey",
+      "image": "Honey",
       "tier": 1,
       "attack": 1,
       "health": 1
     },
     {
       "name": "pill",
       "display": "Sleeping Pill",
+      "image": "Sleeping_Pill",
```
Comment (prindle-jason) [inline src/assets/library.json]: sleepingPill.webp

File: src/views/components/DisplayCard.vue [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+    name: 'DisplayCard',
+    props: {
+        asset: {
+            type: Object,
+            required: true,
+        },
+        type: {
+            type: String,
+            required: true,
+            description: 'pet/shopPet/shopItem',
```
Comment (prindle-jason) [inline src/views/components/DisplayCard.vue:27]: image

Comment (prindle-jason) [reply src/views/components/DisplayCard.vue:27]: I don't have an actual change in mind-- I just dislike this.

File: src/views/components/DisplayCard.vue [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+            return path;
+        },
+        getCardStyle: function () {
+            return 'width:' + this.cardWidth + 'rem;height:' + this.cardHeight + 'rem';
+        },
+        getXpPath: function () {
+            return '/src/assets/images/misc/xp' + this.asset.xp + '.png';
+        },
+        getTierPath: function () {
+            return '/src/assets/images/misc/Tier_' + this.asset.tier + '_Icon.webp';
```
Comment (prindle-jason) [inline src/views/components/DisplayCard.vue]: I had renamed the tier icons to something like "iconTier1".

Were you on the latest code, or did I not push that, or was it all a fever dream?

**Review by prindle-jason — DISMISSED** [review]

File: src/assets/library.json (hunk 2 above)
Comment (shortridge11) [reply src/assets/library.json]: ?
image

File: src/assets/library.json (hunk 1 above)
Comment (shortridge11) [reply src/assets/library.json]: fixed

File: src/assets/library.json (hunk 3 above)
Comment (shortridge11) [reply src/assets/library.json]: fixed

File: src/assets/library.json (hunk 4 above)
Comment (shortridge11) [reply src/assets/library.json]: i guess half the files here took that change? lol
image

Comment (shortridge11) [reply src/assets/library.json]: changed all images to camelcase on this branch

File: src/views/components/DisplayCard.vue (hunk 6 above)
Comment (shortridge11) [reply src/views/components/DisplayCard.vue]: idk, i'm going to push what i have fixed from this stuff but i rebased on main and the only files that were renamed were the ones with underscores in them

File: src/views/components/DisplayCard.vue (hunk 5 above)
Comment (shortridge11) [reply src/views/components/DisplayCard.vue:27]: yea... it made it a ton easier to display things when i redid stuff

Comment (shortridge11) [reply src/views/components/DisplayCard.vue:27]: i think this is really the only real issue left, imo pull my branch and take a look at this file and see how simplified displaying everything is

**Review by jurramonga — APPROVED** [review]

**Pull request merged**